client/src/api/results.js

```diff
@@ -1,148 +1,124 @@
-export async function fetchResults() {
-  // TODO: Connect API here later
-  /*
-  const response = await fetch("localhost:3001/app/offers", {
-    header: {
-      //TODO:
-    },
-  });*/
+export async function fetchAvailableOffers() {
+  try {
+    const response = await fetch("/api/offers/available", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    const results = await response.json();
+    console.log(results[0]);
+    return results;
+  } catch (error) {
+    alert(error.message);
+    return null;
```

Comment:
If you catch the error here, it's difficult to react in your UI to specific errors (show error message).
Instead, I would try-catch in your React components / `useEffect`.
Same issue in other fetches.